### PR TITLE
Metric console exporter modified to deal with all known types

### PR DESCRIPTION
**Pull request by cijothomas**
Modifying ConsoleMetricExporter to explore and demo how an actual Exporter would look like.
(will modify both inmemory and console exporters existing today to support metrics as well, once the exporter interfaces are ready)

### CODE REVIEW

File: src/OpenTelemetry/Metrics/Processors/MetricConsoleExporter.cs [hunk 1]
```diff
@@ -34,12 +34,45 @@ public override void OnEnd(MetricItem data)
             {
                 var tags = metric.Attributes.ToArray().Select(k => $"{k.Key}={k.Value?.ToString()}");
 
+                string valueDisplay = string.Empty;
+                if (metric is ISumMetric sumMetric)
+                {
+                    if (sumMetric.Sum is double doubleSum)
+                    {
+                        valueDisplay = ((double)doubleSum).ToString();
```
Comment (reyang) [inline src/OpenTelemetry/Metrics/Processors/MetricConsoleExporter.cs:42]: InvariantCulture?

File: src/OpenTelemetry/Metrics/Processors/MetricConsoleExporter.cs [hunk 2]
```diff
@@ -34,12 +34,45 @@ public override void OnEnd(MetricItem data)
             {
                 var tags = metric.Attributes.ToArray().Select(k => $"{k.Key}={k.Value?.ToString()}");
 
+                string valueDisplay = string.Empty;
+                if (metric is ISumMetric sumMetric)
+                {
+                    if (sumMetric.Sum is double doubleSum)
```
Comment (reyang) [inline src/OpenTelemetry/Metrics/Processors/MetricConsoleExporter.cs:40]: Do we need these if-else conditions or we simply call ToString for whatever point types?

**Review by reyang — APPROVED** [review]
LGTM with minor suggestions.

File: src/OpenTelemetry/Metrics/Processors/MetricConsoleExporter.cs (hunk 2 above)
Comment (cijothomas) [reply src/OpenTelemetry/Metrics/Processors/MetricConsoleExporter.cs:40]: just to demonstrate that the sum is available as double, if the exporter's backend wants it as double.
(yes in this console, i could just ToString() it.)

**Review by alanwest — APPROVED** [review]
LGTM

File: src/OpenTelemetry/Metrics/Processors/MetricConsoleExporter.cs (hunk 1 above)
Comment (utpilla) [reply src/OpenTelemetry/Metrics/Processors/MetricConsoleExporter.cs]: I think we missed a few more .ToString() calls to add `InvariantCulture`

**Pull request merged**